packages/cli/src/commonlib/sharepointLogin.ts

```diff
@@ -0,0 +1,199 @@
+
+const accountName = "appStudio";
+
+const graphScopes = ["https://graph.microsoft.com/User.ReadBasic.All"];
+const SERVER_PORT = 0;
+
+const graphCachePlugin = new CryptoCachePlugin(accountName);
+const cachePlugin = new CryptoCachePlugin(accountName);
+
+export class SharepointLogin extends login implements SharepointTokenProvider {
+  private static instance: SharepointLogin;
+
+  private static codeFlowInstance: CodeFlowLogin;
+  private graphCodeFlowInstance: CodeFlowLogin;
+
+  private static statusChange?: (
+    status: string,
+    token?: string,
+    accountInfo?: Record<string, unknown>
+  ) => Promise<void>;
+
+  private constructor() {
+    super();
+    this.graphCodeFlowInstance = new CodeFlowLogin(
+      graphScopes,
+      this.getConfig(graphCachePlugin),
+      SERVER_PORT,
+      accountName
+    );
+  }
+
+  /**
+   * Gets instance
+   * @returns instance
+   */
+  public static getInstance(): SharepointLogin {
+    if (!SharepointLogin.instance) {
+      SharepointLogin.instance = new SharepointLogin();
+    }
+
+    return SharepointLogin.instance;
+  }
+
+  /**
+   * Get team access token
+   */
+  async getAccessToken(showDialog = true): Promise<string | undefined> {
+    if (!SharepointLogin.codeFlowInstance) {
+      try {
+        const scopes = await this.getScopes(showDialog);
+        if (!scopes) {
+          return undefined;
+        }
+        SharepointLogin.codeFlowInstance = new CodeFlowLogin(
+          scopes,
+          this.getConfig(cachePlugin),
+          SERVER_PORT,
+          accountName
+        );
+      } catch (error) {
+        throw error;
+      }
+    }
+
+    await SharepointLogin.codeFlowInstance.reloadCache();
+    if (!SharepointLogin.codeFlowInstance.account) {
+      try {
+        const scopes = await this.getScopes(showDialog);
+        if (!scopes) {
+          return undefined;
+        }
+        SharepointLogin.codeFlowInstance.scopes = scopes;
+      } catch (error) {
+        throw error;
+      }
+    }
+    const accessToken = SharepointLogin.codeFlowInstance.getToken();
+    return accessToken;
+  }
+
+  private async getSPTenant(accessToken: string): Promise<string> {
+    const GRAPH_TENANT_ENDPT = "https://graph.microsoft.com/v1.0/sites/root?$select=webUrl";
+
+    if (accessToken.length > 0) {
+      axios.defaults.headers.common.Authorization = `Bearer ${accessToken}`;
+
+      const response = await axios.get(GRAPH_TENANT_ENDPT);
+      return response.data.webUrl;
+    }
+    return "";
+  }
+
+  private async getScopes(showDialog: boolean): Promise<string[] | undefined> {
+    await this.graphCodeFlowInstance.reloadCache();
+    try {
+      const graphToken = await this.getGraphAccessToken(showDialog);
+      if (!graphToken) {
+        return undefined;
+      }
+
+      const tenant = await this.getSPTenant(graphToken!);
+      if (!tenant) {
+        return undefined;
+      }
+      const scopes = [`https://${tenant.split("//")[1]}/Sites.FullControl.All`];
```

Comment:
As tenant begins with https://, can we just use `${tenant}/Sites.FullControl.All`?